src/lxc/criu.c

```diff
@@ -458,8 +466,12 @@ static bool criu_version_ok()
 		if (fgetc(f) != '\n')
 			goto version_error;
 
-		if (strcmp(version, CRIU_VERSION) >= 0)
+		if (strcmp(version, CRIU_VERSION) >= 0) {
+			float tmp = 0;
+			sscanf(version, "%f", &tmp);
+			criu_version = (int)(tmp*100);
```

Comment:
On Tue, Jul 05, 2016 at 07:42:22AM -0700, Adrian Reber wrote:
> > @@ -458,8 +466,12 @@ static bool criu_version_ok()
> >  		if (fgetc(f) != '\n')
> >  			goto version_error;
> >  
> > -		if (strcmp(version, CRIU_VERSION) >= 0)
> > +		if (strcmp(version, CRIU_VERSION) >= 0) {
> > +			float tmp = 0;
> > +			sscanf(version, "%f", &tmp);
> > +			criu_version = (int)(tmp*100);
> 
> Setting `criu_version` only makes sense if the base CRIU version, which is defined with the string `CRIU_VERSION`. If the detected CRIU version is lower we will abort anyway.

Oh, sorry, I misread this as 'goto version_error', you're right that
it's only on version match.

> The float is needed to convert it to an integer value, which sounded easier to me than string comparisons.
> 
> Returning an integer instead of bool in the check function with the CRIU versions sounds like a good idea. It would then be part of criu_opts.
> 
> It is not clear how the conversion from the string with a dot (e.g., "2.3") to an integer should be done without using a float.

`sscanf("%d.%d", &major, &minor)` I think will work for this.

> ---
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/lxc/lxc/pull/1072/files/ff5fee14f4cae14e1ef49b69a889a3ed81e89bab#r69574993